quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java

```diff
@@ -137,15 +137,15 @@ private static Optional<String> resolveUsername(Optional<String> value, ConfigSo
             return of("sa");
         }
 
-        return Database.getDatabaseKind(value.get()).isEmpty() ? value : null;
```

Comment:
I wonder what was the original intention for returning `null` if a username or password matches a DB kind. @pedroigor?

Reply:
It should be an empty optional. The reason why it should not resolve the username is because some databases might not have a default value for this option.